request.go

```diff
@@ -31,17 +31,43 @@ type Response struct {
 // uriForAPI is to be called with something like "/v1/events" and it will give
 // the proper request URI to be posted to.
 func (client *Client) uriForAPI(api string) (string, error) {
-	apiBase, err := url.Parse(client.baseUrl + "/api" + api)
+	var err error
+	// If api is an endpoint starting with v1, we construct the apiBase URL using the baseUrl to which we append "/api" and the actual value of api
+	if strings.Contains(api, "v1") {
```

Comment:
TBH I'm a little unsure about this condition. We could theoretically be passing in full URLs that we expect to just return and which might still contain `v1`. I'd much rather that this was `if !(strings.HasPrefix("https://") or strings.HasPrefix("http://"))`

Reply:
Right, I also started in that direction but both URL types will start with `https`:
- `client.baseUrl + "/api" + api` will be `https://api.datadoghq.com/api/v1....`
- IP ranges URLs: `https://ip-ranges.datadoghq.com/` or `https://ip-ranges.datadoghq.eu/`
So not sure if we should compare based on `https` or `http`?

Reply:
Hey, sorry I'm a bit late here. 
Hmm, this is a good point, because we also aren't sending the API/APP keys in the query params based on this `if` condition. Due to that and the fact that we have some v2 endpoints already (see the dash list v2 endpoint), we can't use this condition, otherwise those requests will be sent without auth and fail. 

Let me know if I'm reading any of this incorrectly 😄 At this point in the code, `api` is either a URI from the caller, like `/v2/dashboards/...` or `/v1/hosts/...` or it's the whole URL for IP ranges `https://ip-ranges.datadoghq.com/`. 
(the `https:api.datadog....` is only added later) So I think the proposed change Slavek made would work here. And then we only add authentication (api/app keys) to paths that are URIs, and not if a full URL is passed into this function.